common4j/src/main/com/microsoft/identity/common/java/util/CommandResultUtil.kt

```diff
@@ -0,0 +1,80 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.util
+
+import com.microsoft.identity.common.java.commands.ICommandResult
+import com.microsoft.identity.common.java.controllers.CommandResult
+
+const val UNSUCCESSFUL_COMMAND_ERROR = "unsuccessful_command"
+
+/**
+ * CommandResult.result value can be of an expected type (ExpectedType), or, if an underlying exception
+ * happened, CommandResult.result is an exception. This method checks for the runtime value of
+ * CommandResult.result and returns the result value if it's of ExpectedType, or it wraps it in an
+ * UnknownError in all other cases.
+ */
+inline fun <reified ExpectedType: com.microsoft.identity.common.java.controllers.results.ICommandResult> CommandResult<Any>.checkAndWrapCommandResultType() : ExpectedType {
```

Comment:
What class is this under? I see the file being called `CommandResult.kt` but I don't see the class defined

Reply:
It's not a class, it's a standalone function.

Reply:
So this is a Kotlin feature that it allows functions to be outside a class? How are these accessed?

Reply:
This is similar to C++ style syntax where functions don't have to be bound to a class https://kotlinlang.org/docs/functions.html#member-functions